### PR TITLE
feat(postgres): adding upgrade scripts to the postgres

### PR DESCRIPTION
**Pull request by duanhongyi**
It is an implementation of this design(#11).

### CODE REVIEW

**Comment by Cryptophobia** [comment]
@till Do you want to give this PR a glance as well?

File: rootfs/bin/do_upgrade
Comment (Cryptophobia) [inline rootfs/bin/do_upgrade:45]: This is nicely done. Good work! :100:

File: rootfs/docker-entrypoint.sh
Comment (Cryptophobia) [inline rootfs/docker-entrypoint.sh:105]: Likewise, great addition to the docker-entrypoint.sh

**Review by Cryptophobia — APPROVED** [review]
lgtm

**Comment by Cryptophobia** [comment]
Looks good to me and I am ready to merge. Anybody else want to give it a glance?

@duanhongyi have you tested the upgrade process with different versions?

**Comment by till** [comment]
@Cryptophobia Yeah, I can have a look after lunch, thanks for the ping!

File: rootfs/bin/do_upgrade
Comment (till) [inline rootfs/bin/do_upgrade:22]: Should this fail/stop when it's neither of the above?

**Comment by Cryptophobia** [comment]
Thanks @till ! Did you find anything that may be a problem in the future that may block this merge?

**Review by till — COMMENTED** [review]
Minor question, otherwise, this looks good.

File: rootfs/bin/do_upgrade
Comment (Cryptophobia) [reply rootfs/bin/do_upgrade:22]: Interesting question. I was thinking about the same but since this script only updates the existing in-cluster postgresDB, that version should never be below 9.4 since that is what came with previous versions of Deis Workflow pre-v1.19.0. So if it is ever a version below 9.4 then the user must have done something very custom. It will be on the user to figure out how to do that upgrade....

Is my thinking correct @duanhongyi ?

Comment (till) [reply rootfs/bin/do_upgrade:22]: Wouldn't hurt to just "die" and echo something like, "We don't support upgrades from X.Y.Z" if there is an unexpected version. Especially as the script continues, as far as I can see. ;)

Comment (Cryptophobia) [reply rootfs/bin/do_upgrade:22]: Sounds good to me. @duanhongyi do you have some time to add an echo and an exit command if the version is unsupported?

Comment (duanhongyi) [reply rootfs/bin/do_upgrade:22]: @Cryptophobia  
I'm on vacation and I'm expected to return at the weekend.

Comment (Cryptophobia) [reply rootfs/bin/do_upgrade:22]: @duanhongyi , sorry to bother you and enjoy your vacation! We can merge this in when you are back. :+1:

Comment (duanhongyi) [reply rootfs/bin/do_upgrade:22]: @Cryptophobia @till Should this fail/stop when it's neither of the above. 

```
else
  echo "Upgrading from "${POSTGRES_OLD_VERSION}" is not supported"
  exit 1
fi
```

Comment (Cryptophobia) [reply rootfs/bin/do_upgrade:22]: Maybe it should do things in this order:
``` 
1. echo a message 
2. timeout and wait 2-3 minutes 
3. exit 1 
```

I think it should fail but with a very clear log message. Then the user should go back to using an older `hephy/postgres` image that does not try to invoke a script to do the upgrade. :+1: 

Any thoughts @till ?

**Review by Cryptophobia — APPROVED** [review]
lgtm

**Comment by Cryptophobia** [comment]
Merged. Thank you for the contribution @duanhongyi !

**Pull request merged**
